spec/system/provider_interface/application_as_pdf_spec.rb

```diff
@@ -0,0 +1,78 @@
+require 'rails_helper'
+
+RSpec.feature 'Provider views an application as PDF' do
+  include CourseOptionHelpers
+  include DfESignInHelpers
+
+  scenario 'viewing application in PDF format' do
+    given_i_am_a_provider_user_with_dfe_sign_in
+    and_i_am_permitted_to_see_applications_for_my_provider
+    and_i_sign_in_to_the_provider_interface
+    and_an_application_exists
+
+    when_i_visit_the_provider_application_page
+    then_i_can_see_a_print_this_page_link
+    and_i_click_the_pdf_link
+    then_i_should_see_the_application_choice_in_pdf_format
+  end
+
+  def given_i_am_a_provider_user_with_dfe_sign_in
+    provider_exists_in_dfe_sign_in
+  end
+
+  def and_i_am_permitted_to_see_applications_for_my_provider
+    provider_user_exists_in_apply_database
+  end
+
+  def and_an_application_exists
+    current_provider = create(:provider, :with_signed_agreement, code: 'ABC')
+    @course_option = course_option_for_provider(
+      provider: current_provider,
+      course: create(:course, name: 'Alchemy', provider: current_provider),
+    )
+
+    @application_form = create(:application_form, first_name: 'Sheila', last_name: 'Jones')
+
+    @application_choice = create(
+      :application_choice,
+      :awaiting_provider_decision,
+      course_option: @course_option,
+      status: 'offer',
+      application_form: @application_form,
+      offered_at: 1.day.ago,
+      updated_at: 1.day.ago,
+    )
+  end
+
+  def when_i_visit_the_provider_application_page
+    visit provider_interface_application_choice_path(@application_choice.id)
+  end
+
+  def then_i_can_see_a_print_this_page_link
+    expect(page).to have_link('Print this page', href: 'javascript:window.print();')
+  end
+
+  def and_i_click_the_pdf_link
+    click_on 'View as PDF'
+  end
+
+  def then_i_should_see_the_application_choice_in_pdf_format
+    expect(page.driver.response.status).to eq(200)
+    expect(page.driver.response.content_type).to eq('application/pdf')
+
+    convert_pdf_to_page
+
+    expect(page).to have_content(@application_form.support_reference)
+    expect(page).to have_content(@application_form.full_name)
+    expect(page).to have_content(@course_option.course.name_and_code)
+    expect(page).not_to have_link('View as PDF')
+  end
+
+  def convert_pdf_to_page
+    temp_pdf = Tempfile.new('pdf')
+    temp_pdf << page.source.force_encoding('UTF-8')
+    reader = PDF::Reader.new(temp_pdf)
+    pdf_text = reader.pages.map(&:text)
+    page.driver.response.instance_variable_set('@body', pdf_text)
```

Comment:
This is witchcraft 😂

Reply:
It's also failing on CI, which makes me think it's probably not a reliable way of testing the response body. I'll do something simpler.